flow/peerdbenv/dynamicconf.go

```diff
@@ -354,3 +379,11 @@ func PeerDBIntervalSinceLastNormalizeThresholdMinutes(ctx context.Context, env m
 func PeerDBApplicationNamePerMirrorName(ctx context.Context, env map[string]string) (bool, error) {
 	return dynamicConfBool(ctx, env, "PEERDB_APPLICATION_NAME_PER_MIRROR_NAME")
 }
+
+func PeerDBMaintenanceModeEnabled(ctx context.Context, env map[string]string) (bool, error) {
+	return dynamicConfBool(ctx, env, "PEERDB_MAINTENANCE_MODE_ENABLED")
+}
+
+func UpdatePeerDBMaintenanceModeEnabled(ctx context.Context, pool *pgxpool.Pool, enabled bool) error {
+	return UpdateDynamicSetting(ctx, pool, "PEERDB_MAINTENANCE_MODE_ENABLED", ptr.String(strconv.FormatBool(enabled)))
```

Comment:
If this is controlled by system I'm not sure we want to expose it to user where they can toggle it

Reply:
This can be helpful for manual intervention or maintenance. We can hide it later from dynamic config if needed

Reply:
Yes, maybe via oauth where different roles can map to different permissions like superadmin, admin or readonly